colossalai/tensor/sharding_spec.py

```diff
@@ -43,24 +43,43 @@ def __repr__(self):
             target += str(dim)
         return target
 
-    def _convert_str_to_shard_list(self, str_spec):
+    @property
+    def difference_dict(self):
         """
-        Convert str_spec into shard_list.
+        Returns the difference dict, and lazily initializes it when needed
 
-        Argument:
-            str_spec(str): dim spec in str type.
+        Return:
+            difference_dict(Dict[Tuple[int, int], Union[int, float, str]]):
+                difference dict
         """
+        if self._DIFFERENCE_DICT is None:
+            self._DIFFERENCE_DICT = self._build_difference_2d_dict()
 
-        if str_spec == "R":
-            return []
-        if str_spec == "S0":
-            return [0]
-        if str_spec == "S1":
-            return [1]
-        if str_spec == "S01":
-            return [0, 1]
+        return self._DIFFERENCE_DICT
 
-    def build_difference_2d_dict(self):
+    def difference(self, other):
+        """
+        The difference between two _DimSpec.
+
+        Argument:
+            other(_DimSpec): the dim spec to compare with.
+
+        Return:
+            difference(int): the difference between two _DimSpec.
+
+        Example:
+            dim_spec = _DimSpec([0])
+            other_dim_spec = _DimSpec([0, 1])
+            print(dim_spec.difference(other_dim_spec))
+
+        Output:
+            5
```

Comment:
This seems to be an old typo. Remove or put "int"?

Reply:
So, should it changed as follows?

```
Output: 
    (int) 5
```

Reply:
Oh I missed it, the return signature is above. So this is actually fine

Reply:
I see. I'll nevertheless push a change to this docstring, because it uses an old class name.